src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,74 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { actions, currentTodoSelector } from '../../features/currentTodo';
+import { useAppSelector } from '../../app/store';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { useDispatch } from 'react-redux';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  const currentTodo = useAppSelector(currentTodoSelector) as Todo;
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    getUser(currentTodo.userId)
+      .then(fetchedUser => {
+        setUser(fetchedUser);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
```

Comment:
Consider handling cases where `getUser` might fail, such as network errors or invalid user IDs. This can be done by adding a `.catch()` block to handle errors gracefully and provide feedback to the user.